host/src/server/api/v1/proof.rs

```diff
@@ -51,16 +52,54 @@ fn set_cached_input(
     };
 
     let path = get_input_path(dir, block_number, network);
-
-    if path.exists() {
-        return Ok(());
-    }
+    info!("caching input for {path:?}");
 
     let file = File::create(&path).map_err(<std::io::Error as Into<HostError>>::into)?;
+    bincode::serialize_into(file, input).map_err(|e| HostError::Anyhow(e.into()))
+}
 
-    info!("caching input for {path:?}");
+async fn generate_input_from_cache(
```

Comment:
Maybe the function should be called `verify_cached_input`?

Reply:
updated.